Source/BugsnagEvent.m

```diff
@@ -430,6 +430,37 @@ - (void)parseHandledCrashData:(NSDictionary *)event {
         return;
     }
     _context = bugsnagPayload[@"context"];
+
+    if (bugsnagPayload[@"metaData"]) {
+        _metadata = [[BugsnagMetadata alloc] initWithDictionary:bugsnagPayload[@"metaData"]];
+    }
+
+    if (bugsnagPayload[@"breadcrumbs"]) {
+        NSArray *crumbs = bugsnagPayload[@"breadcrumbs"];
+        NSMutableArray *data = [NSMutableArray new];
+
+        for (NSDictionary *dict in crumbs) {
+            BugsnagBreadcrumb *crumb = [BugsnagBreadcrumb breadcrumbFromDict:dict];
+            [data addObject:crumb];
+        }
+        _breadcrumbs = data;
+    }
+
+    BOOL unhandled = [bugsnagPayload[@"unhandled"] boolValue];
+    NSDictionary *data = bugsnagPayload[@"severityReason"];
+    BSGSeverity severity = BSGParseSeverity(bugsnagPayload[@"severity"]);
+
+    NSString *attrValue = nil;
+    NSDictionary *attrs = data[@"attributes"];
+
+    if (attrs != nil && [attrs count] == 1) { // only 1 attrValue is ever present
+        attrValue = [attrs allValues][0];
+    }
+    SeverityReasonType reason = [BugsnagHandledState severityReasonFromString:data[@"type"]];
+    _handledState = [[BugsnagHandledState alloc] initWithSeverityReason:reason
+                                                               severity:severity
+                                                              unhandled:unhandled
+                                                              attrValue:attrValue];
```

Comment:
Can you abstract this to a `handledStateFromJson` (or similar) in `BugsnagHandledState` as you have done for app/device/error? Possibly also breadcrumbs?

Reply:
Factored out methods for both breadcrumbs + handled state.